src/naksu/mebroutines/mebroutines.go

```diff
@@ -178,14 +179,18 @@ func RunVBoxManage(args []string) string {
 		fixed, fixErr := detectAndFixDuplicateHardDiskProblem(vBoxManageOutput)
 		if !fixed || fixErr != nil {
 			log.Debug(fmt.Sprintf("Failed to fix problem with command %s (%v)", strings.Join(runArgs, " "), fixErr))
-			showErrorAndQuit(err)
+			if showErrors {
+				showErrorAndQuit(err)
```

Comment:
Olisikohan näissä tapauksissa else-haarassa kuitenkin syytä lopettaa funktion suoritus ja vaikka palauttaa kutsujalle jokin error-arvo, jonka perusteella kutsuja voisi päättää olla käsittelemättä paluuarvoa? Tätä funktiota ei ole kirjoitettu sillä ajatuksella, että suoritus jatkuisi virheen jälkeen.

Pikavilkaisulla tästä voi seurata ainakin seuraavaa:
- "Retrying <command> after fixing problem" logitetaan myös siinä tapauksessa, että duplikaattilevyn korjaus on epäonnistunut (vähän hämmentävää ongelmien debuggauksen kannalta).
- Tuossa alempana olevassa kohdassa `showErrors = false` -tapauksessa päädytään palauttamaan funktion lopussa vboxManageOutput, joka voi olla virhetuloste. Tällöin kutsuja päätyy käsittelemään epävalidia tulostetta, koska ei tiedä virheestä.

Ehdottaisin että muutetaan funktion paluutyypiksi `(string, error)` ja if-rakenteet esim. muotoon

```
if exitOnError {
    showErrorAndQuit(err)
} else {
    return "", err
}
```

Kutsupaikat pitää sitten tietenkin muuttaa käsittelemään virheet oikein.

Reply:
Hyviä ehdotuksia, tein nämä parhaani mukaan.